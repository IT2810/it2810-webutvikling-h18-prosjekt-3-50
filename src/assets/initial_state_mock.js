export default initial_state = {
  sessions: [
    {
      id: 0,
      date: new Date(2018, 9, 2, 13, 40),
      name: 'Session one',
      exercises: [
          { name: 'Squat', sets: '4', reps: '12' },
          { name: 'Benchpress', sets: '4', reps: '12' },
          { name: 'Pullup', sets: '4', reps: '12' },
          { name: 'Row', sets: '4', reps: '12' }
      ],
      contacts: [
          {name: 'Ola Nordmann'},
          {name: 'Kari Hansen'}
      ]
    },
    {
      id: 1,
      date: new Date(2018, 9, 15, 16, 30),
      name: 'Session two',
      exercises: [
          { name: 'Squat', sets: '5', reps: '6' },
          { name: 'Deadlift', sets: '5', reps: '6' },
          { name: 'Pushup', sets: '5', reps: '6' },
          { name: 'Bicepscurl', sets: '5', reps: '6' }
      ],
      contacts: [
          {name: 'Fredrik Hansen'},
          {name: 'Katrine Olavsen'}
      ]
    },
    {
      id: 1,
      date: new Date(2018, 12, 15, 16, 30),
      name: 'Session three',
      exercises: [
          { name: 'Squat', sets: '5', reps: '6' },
          { name: 'Deadlift', sets: '5', reps: '6' },
          { name: 'Pushup', sets: '5', reps: '6' },
          { name: 'Bicepscurl', sets: '5', reps: '6' }
      ],
      contacts: [
          {name: 'Fredrik Hansen'},
          {name: 'Katrine Olavsen'}
      ]
    }
  ],
  selectedDate: new Date(),
  activeSession: {
    contacts: [],
    exercises: []
  }
}