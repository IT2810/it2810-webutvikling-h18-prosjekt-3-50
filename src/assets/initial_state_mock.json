{
  "sessions": [
    {
      "id": 0,
      "date": "2018-10-05T13:40:00.0",
      "name": "Session one",
      "exercises": [
          { "name": "Squat", "sets": "4", "reps": "12" },
          { "name": "Benchpress", "sets": "4", "reps": "12" },
          { "name": "Pullup", "sets": "4", "reps": "12" },
          { "name": "Row", "sets": "4", "reps": "12" }
      ],
      "contacts": [
          {"name": "Ola Nordmann"},
          {"name": "Kari Hansen"}
      ]
    },
    {
      "id": 1,
      "date": "2018-10-19T16:30:00.0",
      "name": "Session two",
      "exercises": [
          { "name": "Squat", "sets": "5", "reps": "6" },
          { "name": "Deadlift", "sets": "5", "reps": "6" },
          { "name": "Pushup", "sets": "5", "reps": "6" },
          { "name": "Bicepscurl", "sets": "5", "reps": "6" }
      ],
      "contacts": [
          {"name": "Fredrik Hansen"},
          {"name": "Katrine Olavsen"}
      ]
    }
  ]
}