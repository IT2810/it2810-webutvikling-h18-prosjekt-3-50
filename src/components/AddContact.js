import React, { Component } from 'react'
import {Â View } from 'react-native'
import { Container, Content, Text, Card, CardItem } from 'native-base';
import { connect } from 'react-redux'


const savedContacts = require('../assets/contacts.json')

import { addContact } from '../../actions/index'

export class AddContact extends Component {
  static navigationOptions = {
    title: 'Choose contact'
  }

  constructor(props, context) {
    super(props, context)
    this.state = {
      availableContacts: savedContacts.contacts
    }

    this._add = this._add.bind(this)
    this.availableContacts = this.availableContacts.bind(this)
  }

  _add(contact) {
    this.props.addContact(contact)
    this.props.navigation.goBack()
  }

  // Filter out the contacts allready added, so that they can't be added twice
  availableContacts() {
    return savedContacts.filter(contact => this.props.contacts.indexOf(contact) === -1)
  }

  render() {
    return (
      <Container>
        <Content>
          <Card>
            {this.state.availableContacts.map((contact, index) => (
              <CardItem
                key={index}
                bordered
                button
                onPress={() => this._add(contact)}
                >
                <Text>
                  { contact.name }
                </Text>
              </CardItem>
            ))}
          </Card>
        </Content>
      </Container>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  addContact: contact => dispatch(addContact(contact))
})

export default connect(null, mapDispatchToProps)(AddContact)