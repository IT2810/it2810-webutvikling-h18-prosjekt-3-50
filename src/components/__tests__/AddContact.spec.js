/**
 * @jest-environment jsdom
 */

import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { AddContact } from '../AddContact'
import 'native-base'
import { findByID } from '../../testUtils.js'

import initial_state_mock from '../../assets/initial_state_mock.js'
import configureStore from 'redux-mock-store'

describe('AddContact', () => {
  const initialState = initial_state_mock
  const mockStore = configureStore()
  let wrapper
  let store
  let dispatchAddContactMock = jest.fn()
  let navigation
  let goBackMock

  beforeEach(() => {
    store = mockStore(initialState)
    goBackMock = jest.fn()
    navigation = {goBack: goBackMock}
    wrapper = shallow(<AddContact navigation={navigation} store={store} addExercise={dispatchAddContactMock} addContact={dispatchAddContactMock} />)
  })

  it('renders correctly', () => {
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

  describe('_add', () => {
    it('calls _add when button is pressed', () => {
      let _addMock = jest.fn()
      let wrapper = shallow(<AddContact navigation={navigation} store={store} _add={_addMock} />)

      wrapper.instance()._add = _addMock

      wrapper.instance().forceUpdate()

      let button = findByID(wrapper, 'addContactButton')
      button.props().onPress()

      expect(_addMock.mock.calls.length).toBe(1)
    })

    it('dispatches addContact', () => {
      var contact = { name: 'Test contact' }
      wrapper.instance()._add(contact)

      expect(dispatchAddContactMock.mock.calls.length).toBe(1)
    })

    it('navigates to CreateSession', () => {
      wrapper.instance()._add()

      expect(goBackMock.mock.calls.length).toBe(1)
    })
  })

})
