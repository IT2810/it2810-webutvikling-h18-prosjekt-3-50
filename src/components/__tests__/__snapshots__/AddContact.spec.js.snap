// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddContact renders correctly 1`] = `
<Container
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <Styled(Content)>
    <Styled(Card)>
      <Styled(CardItem)
        bordered={true}
        button={true}
        key="0"
        onPress={[Function]}
        testID="addContactButton"
      >
        <Styled(Text)>
          Test Name
        </Styled(Text)>
      </Styled(CardItem)>
    </Styled(Card)>
  </Styled(Content)>
</Container>
`;
